### PR TITLE
Reduce time spent in ConflictResolver.Session.GetNodesOrTokensToCheckForConflicts

### PR DESCRIPTION
**Pull request by ToddGrun**
This method previoulsy realized/walked the whole tree to try to find annotations to search for in it's AnnotationTable. Instead, use the GetAnnotatedNodesAndTokens to limit the search to only those areas of the tree that contain annotations.

This showed up in a profile I took of a lightbulb session which showed the preview window.

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 1]
```diff
@@ -159,15 +159,12 @@ public RenameRewriter(RenameRewriterParameters parameters)
 
             var isInConflictLambdaBody = false;
             var lambdas = node.GetAncestorsOrThis(n => n is SimpleLambdaExpressionSyntax or ParenthesizedLambdaExpressionSyntax);
-            if (lambdas.Count() != 0)
+            foreach (var lambda in lambdas)
             {
-                foreach (var lambda in lambdas)
+                if (_conflictLocations.Any(cf => cf.Contains(lambda.Span)))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:164]: no 'any' that takes an arg?

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 2]
```diff
@@ -159,15 +159,12 @@ public RenameRewriter(RenameRewriterParameters parameters)
 
             var isInConflictLambdaBody = false;
             var lambdas = node.GetAncestorsOrThis(n => n is SimpleLambdaExpressionSyntax or ParenthesizedLambdaExpressionSyntax);
-            if (lambdas.Count() != 0)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:162]: YIKES

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs [hunk 3]
```diff
@@ -505,9 +505,13 @@ private async Task DebugVerifyNoErrorsAsync(MutableConflictResolution conflictRe
         private IEnumerable<(SyntaxNodeOrToken syntax, RenameActionAnnotation annotation)> GetNodesOrTokensToCheckForConflicts(
             SyntaxNode syntaxRoot)
         {
-            return syntaxRoot.DescendantNodesAndTokens(descendIntoTrivia: true)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs:508]: YIKES

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs (hunk 1 above)
Comment (ToddGrun) [reply src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs:164]: There isn't one available, and the only current extension methods we have on ImmutableHashSet are in the compiler, and I didn't want to change that in this PR.

**Pull request merged**